enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,391 @@
+tagged and released upstream.
+
+The advantage of managing our own branches is it gives us flexibility in
+keeping master in sync with upstream master so we always have the latest code
+available downstream.
+
+The disadvantage to this is that each release branch is not managed by ART.
+
+## Implementation History
+
+2020-10-13 - Rewrite proposal using the automation option.
+2020-10-07 - Propose to automate the syncing of upstream to downstream.
+2020-09-10 - Propose using the UPSTREAM-MERGE.sh script similar to Service
+             Catalog releases.
+
+## Drawbacks
+
+* We would have to manage our own downstream release branches
+* We would have to handle the automation of updates ourselves
+* The special downstream branch is kind of janky
+
+## Alternatives
+
+### Monorepo
+One alternative is to have a monorepo and sync all upstream repos to a single
+repo. While this is a possible solution it would be more difficult to maintain
+for the needs of the SDK team.
+
+### Semi-manual syncing of repos
+
+Mirror the upstream Operator SDK repos downstream in the openshift GitHub
+organization. We will have a one to one upstream to downstream repo.
+
+* operator-sdk - github.com/openshift/operator-sdk
+* operator-lib - github.com/openshift/operator-lib
+* ocp-release-operator-sdk - to be retired after the above repos are working
+
+The downstream repos will contain specific downstream changes which should be
+additions and non-conflicting:
+
+* Dockerfiles to support CI
+* Dockerfiles to build downstream images
+* Vendor Go dependencies
+* Script to assist with syncing upstream to downstream
+* Ansible collections to support Ansible operator
+
+We would sync `operator-framework/operator-sdk` into `openshift/operator-sdk`
+using a script, [UPSTREAM-MERGE.sh][upstream-merge]
+which we have adapted from the current openshift/service-catalog.
+
+The script is only maintained in master branch. Other branches should copy this
+script into that branch in case newer changes have been made.
+
+The `master` branch maps to a specific OpenShift release. We sync an upstream
+minor or patch release we determined to ship with OpenShift. Once `master`
+switches to a new release, we will reset master to the new release we plan to
+sync for the next OpenShift release. The release branches will continue to be
+updated when necessary the same way.
+
+The process to update `master` will be as follows:
+
+```
+cd operator-sdk
+git checkout master
+./UPSTREAM-MERGE.sh v1.0.0
+# resolve any conflicts; script will put you in a new branch
+git commit
+git push origin NAME_OF_BRANCH
+```
+
+For updating a specific release branch the process will largely look the same:
+
+```
+cd operator-sdk
+git checkout release-4.6 # checkout specific branch to be updated
+./UPSTREAM-MERGE.sh v0.19.5
+# resolve any conflicts; script will put you in a new branch
+git commit
+git push origin NAME_OF_BRANCH
+```
+
+The following is a list of directories and files that we will be adde to
```

Comment:
Is this list duplicated above?

Reply:
Yes it's duplicated because that is the alternative solution in case the automated approach hits a snag, we can go back to the old way.